src/tests/test_jsproxy.py

```diff
@@ -508,3 +502,205 @@ def test_register_jsmodule_docs_example(selenium):
+            result.append(["has_get1", obj.has_get[10], 10])
+            result.append(["has_get2", obj.has_get[11], 11])
+            m = obj.has_getset
+            m[1] = 6
+            m[2] = 77
+            m[3] = 9
+            m[2] = 5
+            del m[3]
+            result.append(["has_getset", [x.to_py() for x in m.entries()], [[1, 6], [2, 5]]])
+            result.append(["has_has", [n in obj.has_has for n in ["x9", "a9"]], [True, False]])
+            result.append(["has_includes", [n in obj.has_includes for n in ["x9", "a9"]], [False, True]])
+            result.append(["has_has_includes", [n in obj.has_has_includes for n in ["x9", "a9"]], [True, False]])
+            result.append(["awaitable", await obj.awaitable, 7])
+            result
+        `);
+        return result.toJs();
+        """
+    )
+    for [desc, a, b] in result:
+        assert a == b, desc
+
+
+def test_mixins_errors(selenium):
+    selenium.run_js(
+        """
+        window.a = [];
+        window.b = { 
+            has(){ return false; },
+            get(){ return undefined; },
+            set(){ return false; },
+            delete(){ return false; },
+        };
+        await pyodide.runPythonAsync(`
+            from pytest import raises
+            from js import a, b
+            with raises(IndexError):
+                a[0]
+            with raises(IndexError):
+                del a[0]
+            with raises(KeyError):
+                b[0]
+            with raises(KeyError):
+                del b[0]
+        `);
+
+        window.c = {
+            next(){},
+            length : 1,
+            get(){},
+            set(){},
+            has(){},
+            then(){}     
+        };
+        window.d = {
+            [Symbol.iterator](){},
+        };
+        pyodide.runPython("from js import c, d");
+        delete c.next;
+        delete c.length;
+        delete c.get;
+        delete c.set;
+        delete c.has;
+        delete c.then;
+        delete d[Symbol.iterator];
+        await pyodide.runPythonAsync(`
+            from pytest import raises
+            from pyodide import JsException
+            msg = "^TypeError:.* is not a function$"
+            with raises(JsException, match=msg):
+                next(c)
+            with raises(JsException, match=msg):
+                iter(d)
+            with raises(TypeError, match="object does not have a valid length"):
+                len(c)
+            with raises(JsException, match=msg):
+                c[0]
+            with raises(JsException, match=msg):
+                c[0] = 7
+            with raises(JsException, match=msg):
+                del c[0]
+            with raises(TypeError, match="can't be used in 'await' expression"):
+                await c
+        `);
+
+        window.l = [0, false, NaN, undefined, null];
+        window.l[6] = 7;
+        await pyodide.runPythonAsync(`
+            from js import l
+            from pytest import raises
```

Comment:
Would be nice to have a shim since `pytest.raises` functions like both `assertRaises` and `assertRaisesRegex`. But I don't think that `testing.py` is a good place for it: as much as possible I want tests to run when copied into the browser console in `console.html`. For now I can just duplicate it a bunch.